app/code/Magento/SalesRule/Model/Quote/Discount.php

```diff
@@ -119,7 +130,20 @@ public function collect(
 
         $address->setDiscountDescription([]);
         $items = $this->calculator->sortItemsByPriority($items, $address);
-        $address->getExtensionAttributes()->setDiscounts([]);
+
+        if (!is_object($address->getExtensionAttributes())) {
```

Comment:
Hi,
Looks like this is collect totals and it shouldn’t change type of extension attributes, but getExtensionAttrubutes should return always object. I think shipping address object should come just with correct extension attributes, so place that prepares incorrect object should be adjusted instead
What do you think?

Reply:
> getExtensionAttrubutes should return always object

yes, true

Reply:
Seems problem in private method **extractAddressData**
And solution by @amenk working.

I reverted own changes in Discount class.